### PR TITLE
Declare DSL extensions as direct functions to simplify usage.

### LINKED ISSUE #284: Replace DSL Extensions with Public Methods
It is kind of annoying to import DSL operators semi-manually. Yes, the IDE most of the time will suggest the right thing, but it is still confusing to see DSL operators as “optional” extensions. Technically it is nice to see how it is done via extensions, the real world usability is not so great. If we all agree about DSL stability across releases this shouldn’t be an issue.

Pros.
* Less semi-manual imports which take seconds every time you write new test and is very annoying.
* Better adoption rate because of less confusion “what to import”.

Cons.
* None?

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #284.

### CODE REVIEW

File: spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt [hunk 1]
```diff
@@ -25,4 +25,92 @@ interface SpecBody: TestContainer {
...
+
+    /**
+     * Creates a [group][SpecBody.group].
+     *
+     * @author Ranie Jade Ramiso
+     * @since 1.0
+     */
+    @Synonym(type = SynonymType.Group, prefix = "describe")
+    fun describe(description: String, body: SpecBody.() -> Unit) {
+        group("describe $description", body = body)
```
Comment (arturdryomov) [inline spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt:37]: It’s a bummer that we cannot reuse `prefix` from the annotation here :-(

File: spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt [hunk 2]
```diff
@@ -25,4 +25,92 @@ interface SpecBody: TestContainer {
...
+    }
+
+    /**
+     * Creates a [group][SpecBody.group].
+     *
+     * @author Ranie Jade Ramiso
+     * @since 1.0
```
Comment (arturdryomov) [inline spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt:44]: Maybe remove these?

**Review by arturdryomov — APPROVED** [review]

File: spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt (hunk 2 above)
Comment (artem-zinnatullin) [reply spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt:44]: I would remove that, but it's @raniejade's child, so only if he agrees :)

File: spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt (hunk 1 above)
Comment (artem-zinnatullin) [reply spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt:37]: Not a bummer, but yeah ¯\_(ツ)_/¯

**Review by raniejade — APPROVED** [review]
nit: Should we remove the author tags? They become unmaintained as project the goes on. (it was an old habit of mine)

File: spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt (hunk 2 above)
Comment (raniejade) [reply spek-dsl/common/src/main/kotlin/org/spekframework/spek2/dsl/SpecBody.kt:44]: :+1: actually I just commented about it :).

**Comment by artem-zinnatullin** [comment]
@raniejade #307, will address in a separate PR

**Pull request merged**